content/en/docs/advanced-tasks/catalog-update-formulary.md

```diff
@@ -0,0 +1,382 @@
+  - image: quay.io/organization/testoperator:v1.0.1
+  - image: quay.io/organization/testoperator:v1.1.0
+  - image: quay.io/organization/testoperator:v1.1.1
+...
+```
+
+#### FBC
+Add a new `testoperator.v1.1.1` edge to an existing `candidate-v1.1` channel
+
+```bash
+yq eval 'select(.schema=="olm.channel" and .name == "candidate-v1.1").entries += [{"name" : "testoperator.v1.1.1"}]' fbc.yaml
+```
+
+produces updated `candidate-v1.1` channel:
+
+```yaml
+...
+entries:
+  - name: testoperator.v1.1.0
+    replaces: testoperator.v1.0.1
+    skips:
+      - testoperator.v1.0.0
+  - name: testoperator.v1.1.1
+name: candidate-v1.1
+package: testoperator
+schema: olm.channel
+...
+```
+
+
+### Adding a new 'replaces' link between two existing bundles
+
+#### FBC
+Adding a new `testoperator.v1.1.1` bundle version edge with a replaces link to its predecessor `testoperator.v1.1.0` version which already exists in the channel. 
+
+```bash
+yq eval 'select(.schema == "olm.channel" and .name == "candidate-v1.1").entries += [{"name" : "testoperator:v1.1.1", "replaces": "testoperator:v1.1.0"}]' fbc.yaml
+```
+produces updated `candidate-v1.1` channel:
+
+```yaml
+...
+entries:
+  - name: testoperator.v1.1.0
+    replaces: testoperator.v1.0.1
+    skips:
+      - testoperator.v1.0.0
+  - name: testoperator:v1.1.1
+    replaces: testoperator:v1.1.0
+name: candidate-v1.1
+package: testoperator
+schema: olm.channel
+...
+```
+
+### Removing a specific bundle version
+
+#### FBC
+Remove the upgrade edge from the example `candidate-v1.1` channel which refers to bundle version `testoperator.v1.1.0`. 
+
+```bash
+yq eval 'del(select(.schema == "olm.channel" and .name == "candidate-v1.1" ).entries[]| select(.name == "testoperator.v1.1.0"))' fbc.yaml
+```
+
+produces updated `candidate-v1.1` channel:
+
+```yaml
+...
+entries: []
+name: candidate-v1.1
+package: testoperator
+schema: olm.channel
+...
+```
+Please note that removing the only edge for a channel as in this example will yield an explicitly empty array.  This will produce an error in `opm validate`. 
+
+### Substituting a bundle version in the upgrade graph
+
+#### semver
+For all channels, replace instances of `quay.io/organization/testoperator:v1.1.0` with `quay.io/organization/testoperator:v1.1.0-CVE`
+
+```bash
+yq '(..| select(has("image") and .image == "quay.io/organization/testoperator:v1.1.0")).image = "quay.io/organization/testoperator:v1.1.0-cve"' semver.yaml
+```
+produces updated template:
+
+```yaml
+schema: olm.semver
+generatemajorchannels: false
+generateminorchannels: true
+candidate:
+  bundles:
+    - image: quay.io/organization/testoperator:v1.0.0
+    - image: quay.io/organization/testoperator:v1.0.1
+    - image: quay.io/organization/testoperator:v1.1.0-cve
+fast:
+  bundles:
+    - image: quay.io/organization/testoperator:v1.0.1
+    - image: quay.io/organization/testoperator:v1.1.0-cve
+stable:
+  bundles:
+    - image: quay.io/organization/testoperator:v1.0.1
+```
+
+#### FBC
```

Comment:
For this, the correct freshmaker style algorithm is here: https://github.com/operator-framework/operator-registry/issues/782#issuecomment-931411205